File: tests/fixtures/cpp/globals.cpp
#include <cstdint>

volatile std::int32_t global_sink;

namespace fixture {
inline volatile std::int32_t inline_value = 111;
static volatile std::int32_t namespace_static = 112;

namespace alpha {
volatile std::int32_t duplicate = 121;
}

namespace beta {
volatile std::int32_t duplicate = 122;
}

namespace {
volatile std::int32_t anonymous_value = 123;
}

struct Holder {
    static volatile std::int32_t member;
    inline static volatile std::int32_t inline_member = 132;
    static constexpr std::int32_t constexpr_member = 133;
    static constexpr std::int32_t negative_constexpr_member = -123;
};

volatile std::int32_t Holder::member = 131;
}  // namespace fixture

__attribute__((noinline)) void inspect_globals() {
    global_sink = fixture::inline_value + fixture::namespace_static +
                  fixture::alpha::duplicate + fixture::beta::duplicate +
                  fixture::anonymous_value + fixture::Holder::member +
                  fixture::Holder::inline_member +
                  fixture::Holder::constexpr_member +
                  fixture::Holder::negative_constexpr_member;
}

int main() {
    inspect_globals();
    return global_sink == 862 ? 0 : 1;
}

File: tests/fixtures/zig/globals.zig
const std = @import("std");

pub var root_value: i64 = 171;
pub const root_constant: i64 = 172;
pub var pointer_value: i32 = 184;
pub var root_pointer: *const i32 = &pointer_value;
var global_sink: i64 = 0;

const Alpha = struct {
    pub var duplicate: i32 = 181;
    pub const constant: i32 = 182;
};

const Beta = struct {
    pub var duplicate: i32 = 183;
};

noinline fn inspectGlobals() void {
    global_sink = root_value + root_constant + Alpha.duplicate +
        Alpha.constant + Beta.duplicate + root_pointer.*;
    std.mem.doNotOptimizeAway(&global_sink);
}

pub fn main() u8 {
    inspectGlobals();
    return @intFromBool(global_sink != 1073);
}
